pkg/ipamd/datastore/data_store.go

```diff
@@ -266,9 +268,27 @@ func (addr AddressInfo) Assigned() bool {
 	return !addr.IPAMKey.IsZero()
 }
 
+var log = logger.Get()
+
+// getCoolingPeriod returns the time duration in seconds configured by ADDRESS_COOLING_PERIOD env variable
+func getCoolingPeriod() time.Duration {
+	inputStr, found := os.LookupEnv(envAddressCoolingPeriod)
+	if !found {
+		return defaultAddressCoolingPeriod
+	}
+	if input, err := strconv.Atoi(inputStr); err == nil {
+		if input < 0 {
+			return defaultAddressCoolingPeriod
+		}
+		log.Debugf("Using ADDRESS_COOLING_PERIOD %v", input)
+		return time.Duration(input) * time.Second
+	}
+	return defaultAddressCoolingPeriod
+}
+
 // InCoolingPeriod checks whether an addr is in addressCoolingPeriod
 func (addr AddressInfo) inCoolingPeriod() bool {
-	return time.Since(addr.UnassignedTime) <= addressCoolingPeriod
+	return time.Since(addr.UnassignedTime) <= getCoolingPeriod()
```

Comment:
nit: Set it during DataStore construction and then you can access a member variable instead of having the overhead of a function call each time

Reply:
I'm not sure about this one. If addressCoolingPeriod is set as a member variable of Datastore during NewDataStore, then it would not be available while calling (addr AddressInfo) inCoolingPeriod. Alternatively, addressCoolingPeriod could be a normal variable in the package scope.